test/tests/docker-registry-push-pull/run.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+set -eo pipefail
+
+dir="$(dirname "$(readlink -f "$BASH_SOURCE")")"
+
+image="$1"
+case "${image##*/}" in
+	docker:*dind*)
+		dockerImage="$image"
+		registryImage='registry'
+		;;
+	registry:*|registry)
+		registryImage="$image"
+		dockerImage='docker:dind'
+		;;
+	*)
+		echo >&2 "error: unable to determine whether '$image' is registry or docker:dind"
+		exit 1
+		;;
+esac
+
+rhostname='reg.example.com'
+rnamespace="${rhostname}:5000"
+
+rcname="docker-registry-container-$RANDOM-$RANDOM"
+rcid="$(
+	docker run -d -it \
+		--hostname "$rhostname" \
+		--name "$rcname" \
+		"$registryImage"
+)"
+trap "docker rm -vf $rcid > /dev/null" EXIT
+
+dcname="docker-daemon-container-$RANDOM-$RANDOM"
+dcid="$(
+	docker run -d -it \
+		--privileged \
+		--link "$rcid":"$rhostname" \
```

Comment:
@aaronlehmann aww, I knew it was discouraged, but I didn't realize we'd gone all the way to deprecated already :disappointed:

We've got a lot of tests using `--link` ATM (really useful for making a simple test that doesn't require extra networking setup / cleanup), so I think we'll probably want to figure out a more general solution to get rid of `--link` across our whole test suite (so I think this is OK for now since it's consistent with our other tests, but good to know we need to start thinking more aggressively about updating our tests and docs). :heart: